src/backoffice/shared/components/tree/tree-item-base/tree-item-base.element.ts

```diff
@@ -0,0 +1,174 @@
+			this.observe(this.#treeItemContext.isLoading, (value) => (this._isLoading = value));
+			this.observe(this.#treeItemContext.isSelectable, (value) => (this._isSelectable = value));
+			this.observe(this.#treeItemContext.isSelected, (value) => (this._isSelected = value));
+			this.observe(this.#treeItemContext.hasActions, (value) => (this._hasActions = value));
+			this.observe(this.#treeItemContext.path, (value) => (this._href = value));
+		});
+	}
+
+	connectedCallback(): void {
+		super.connectedCallback();
+		this.addEventListener('selected', this._handleSelectedItem);
+		this.addEventListener('unselected', this._handleDeselectedItem);
+	}
+
+	private _handleSelectedItem(event: Event) {
+		event.stopPropagation();
+		this.#treeItemContext?.select();
+	}
+
+	private _handleDeselectedItem(event: Event) {
+		event.stopPropagation();
+		this.#treeItemContext?.deselect();
+	}
+
+	// TODO: do we want to catch and emit a backoffice event here?
+	private _onShowChildren() {
+		if (this._childItems && this._childItems.length > 0) return;
+		this.#observeChildren();
+	}
+
+	async #observeChildren() {
+		if (!this.#treeItemContext?.requestChildren) return;
+
+		const { asObservable } = await this.#treeItemContext.requestChildren();
+		if (!asObservable) return;
+
+		this.observe(asObservable(), (childItems: any) => {
+			this._childItems = childItems;
+		});
+	}
+
+	private _openActions() {
+		this.#treeItemContext?.toggleContextMenu();
+	}
+
+	render() {
+		return html`
+			<uui-menu-item
+				@show-children=${this._onShowChildren}
+				?selectable=${this._isSelectable}
+				?selected=${this._isSelected}
+				.loading=${this._isLoading}
+				.hasChildren=${this._hasChildren}
+				label="${ifDefined(this.item?.name)}"
+				href="${ifDefined(this._href)}">
+				${this.#renderIcon()} ${this.#renderLabel()} ${this.#renderActions()} ${this.#renderChildItems()}
+				<slot></slot>
+			</uui-menu-item>
+		`;
+	}
+
+	#hasNodes = (e: Event) => {
+		return (e.target as HTMLSlotElement).assignedNodes({ flatten: true }).length > 0;
+	};
+
+	#renderIcon() {
+		return html`
+			<slot
+				name="icon"
+				slot="icon"
+				@slotchange=${(e: Event) => {
+					this._iconSlotHasChildren = this.#hasNodes(e);
+				}}></slot>
+			${this.item?.icon && !this._iconSlotHasChildren
+				? html` <uui-icon slot="icon" name="${this.item.icon}"></uui-icon> `
+				: nothing}
+		`;
+	}
+
+	#renderLabel() {
+		return html`<slot name="label" slot="label"></slot>`;
+	}
+
+	#renderActions() {
+		return html`
+			${this._hasActions
+				? html`
+						<uui-action-bar slot="actions">
+							<uui-button @click=${this._openActions} label="Open actions menu">
+								<uui-symbol-more></uui-symbol-more>
+							</uui-button>
+						</uui-action-bar>
+				  `
+				: nothing}
+		`;
+	}
+
+	#renderChildItems() {
+		return html`
+			${this._childItems
+				? repeat(
+						this._childItems,
+						// TODO: get unique here instead of name. we might be able to get it from the context
```

Comment:
Ouch, Just for rendering/identification of data changes. I think we should request backend to reveal the key/unique for the data? Otherwise re-rendering/RxJS updates cannot be optimized?

Reply:
Seems alright for now.